src/Dict/Extra.elm

```diff
@@ -243,6 +249,46 @@ insertCombining combine key value dict =
     Dict.update key with dict
 
 
+{-| Updates a value if the key is present in the dictionary, leaves the dictionary untouched otherwise.
+
+    import Dict
+
+    Dict.fromList [ ( "expenses", 38.25 ), ( "assets", 100.85 ) ]
+        |> updateIfExists "expenses" ((+) 2.50)
+        |> updateIfExists "liabilities" ((-) 2.50)
+        --> Dict.fromList [ ( "expenses", 40.75 ), ( "assets", 100.85 ) ]
+
+-}
+updateIfExists : comparable -> (a -> a) -> Dict comparable a -> Dict comparable a
+updateIfExists key f dict =
+    case Dict.get key dict of
+        Just value ->
+            Dict.insert key (f value) dict
+
+        Nothing ->
+            dict
+
+
+{-| Updates a value if the key is present in the dictionary, inserts a new key-value pair otherwise.
+
+    import Dict
+
+    Dict.fromList [ ( "expenses", 38.25 ), ( "assets", 100.85 ) ]
+        |> upsert "expenses" 4.50 ((+) 2.50)
+        |> upsert "liabilities" 2.50 ((-) 2.50)
```

Comment:
I guess it doesn't matter in this example but in general `(-)` might be confusing for folks. So I'd just use a lambda for that here